src/HexagonOptimize.cpp

```diff
@@ -391,6 +397,17 @@ class OptimizePatterns : public IRMutator {
             if (neg_b.defined()) {
                 expr = mutate(op->a + neg_b);
                 return;
+            } else {
+                static vector<Pattern> subs = {
+                    // Widening subtracts. There are other instrucitons that subtact two vub and two vuh but do not widen.
```

Comment:
instructions*

It also seems like this comment belongs on the intrinsics definition in CodeGen_Hexagon.cpp rather than here? Same for the add.